### PR TITLE
More EA redesign improvements

### PR DESCRIPTION
**Pull request by oetherington**
Changes in this PR:

- Limit post titles to a single line
Screenshot 2023-03-20 at 23 06 22

- Make comments expandable (this required an extra level of div nesting, so the diff looks a lot worse than it actually is)
Screenshot 2023-03-20 at 23 06 43

- Fix the casing and date format on the all posts page (requested by Lizka)
Screenshot 2023-03-21 at 12 30 20
Screenshot 2023-03-21 at 12 30 10

- Sans serif for sunshine sidebar post titles
Screenshot 2023-03-20 at 23 06 59

- Add a hyphen to "inflation-adjusted" (as requested by Lizka)
Screenshot 2023-03-21 at 12 21 41

- Fix a bug where curated posts could be shown twice in the posts list

### CODE REVIEW

**Comment by agnesstenlund** [comment]
this is so great @oetherington 🥲 👏

**Review by s-cheng — APPROVED** [review]
Looks great! 😄 Just need to change one thing

File: packages/lesswrong/components/posts/EAPostsItem.tsx [hunk 1]
```diff
@@ -73,16 +78,11 @@ export const styles = (theme: ThemeType): JssStyles => ({
     fontSize: 16,
     fontFamily: theme.palette.fonts.sansSerifStack,
     marginBottom: 2,
-    display: "-webkit-box",
-    "-webkit-box-orient": "vertical",
-    "-webkit-line-clamp": 2,
-    [theme.breakpoints.down("xs")]: {
-      "-webkit-line-clamp": 3,
-    },
```
[the remark this answers is not in the document]
Comment (oetherington) [reply packages/lesswrong/components/posts/EAPostsItem.tsx]: This should now be fixed! Thanks!

[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/components/posts/EAPostsItem.tsx]: @oetherington I see you changed it to truncate at 2 lines, but shouldn't it be 3?

**Pull request merged**